### PR TITLE
auto identify differential pairs

### PR DESCRIPTION
**Pull request by ring630**

### CODE REVIEW

**Comment by ring630** [comment]
unittest is added for all the new methods. I don't know why it is still failing codecov.

**Comment by ring630** [comment]
> ## [Codecov](https://app.codecov.io/gh/ansys/pyaedt/pull/3394?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ansys) Report
> > Merging [#3394](https://app.codecov.io/gh/ansys/pyaedt/pull/3394?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ansys) ([dcb1e71](https://github.com/ansys/pyaedt/commit/dcb1e71da4a4cccacb2b6e87cfc9a22deb99afe7)) into [main](https://app.codecov.io/gh/ansys/pyaedt/commit/b43955fbd036f8f224b79b475f4d7459b6b0adad?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ansys) ([b43955f](https://github.com/ansys/pyaedt/commit/b43955fbd036f8f224b79b475f4d7459b6b0adad)) will **decrease** coverage by `0.01%`.
> > The diff coverage is `23.33%`.
> 
> ```diff
> @@            Coverage Diff             @@
> ##             main    #3394      +/-   ##
> ==========================================
> - Coverage   31.54%   31.54%   -0.01%     
> ==========================================
>   Files         175      175              
>   Lines       60272    60300      +28     
> ==========================================
> + Hits        19013    19020       +7     
> - Misses      41259    41280      +21     
> ```


unittest is added for all the new methods. I don't know why it is still failing codecov.

File: pyaedt/edb_core/net_class.py [hunk 1]
```diff
@@ -157,6 +159,74 @@ def create(self, name, net):
...
+            Threshold of resistor value. Search extended net across resistors which has value lower than the threshold.
+        inductor_below : int, float, optional
+            Threshold of inductor value. Search extended net across inductances which has value lower than the
+            threshold.
+        capacitor_above : int, float, optional
+            Threshold of capacitor value. Search extended net across capacitors which has value higher than the
+            threshold.
+        exception_list : list, optional
+            List of components which bypass threshold check. Components in the list are considerred as serial
+        components.The default is ``None``.
```
Comment (PipKat) [inline pyaedt/edb_core/net_class.py:179]:
```suggestion
        resistor_below : int, float, optional
            Threshold for the resistor value. Search the extended net across resistors that have a value lower than the
            threshold.
        inductor_below : int, float, optional
            Threshold for the inductor value. Search the extended net across inductances that have a value lower
            than the threshold.
        capacitor_above : int, float, optional
            Threshold for the capacitor value. Search the extended net across capacitors that have value higher
            than the threshold.
        exception_list : list, optional
            List of components to bypass when performing threshold checks. Components in the list
            are considered as serial components. The default is ``None``.
```

File: pyaedt/edb_core/net_class.py [hunk 2]
```diff
@@ -157,6 +159,74 @@ def create(self, name, net):
...
+
+        Returns
+        -------
+        list
+            List of all extended nets.
```
Comment (PipKat) [inline pyaedt/edb_core/net_class.py:218]:
```suggestion
            List of all extended nets and their associated components.
```

File: pyaedt/edb_core/net_class.py [hunk 3]
```diff
@@ -157,6 +159,74 @@ def create(self, name, net):
...
+            Threshold of resistor value. Search extended net across resistors which has value lower than the threshold.
+        inductor_below : int, float, optional
+            Threshold of inductor value. Search extended net across inductances which has value lower than the
+            threshold.
+        capacitor_above : int, float, optional
+            Threshold of capacitor value. Search extended net across capacitors which has value higher than the
+            threshold.
+        exception_list : list, optional
+            List of components which bypass threshold check. Components in the list are considerred as serial
+        components.The default is ``None``.
```
Comment (PipKat) [inline pyaedt/edb_core/net_class.py:213]:
```suggestion
        resistor_below : int, float, optional
            Threshold for the resistor value. Search the extended net across resistors that have a value
            lower than the threshold.
        inductor_below : int, float, optional
            Threshold for the inductor value. Search the extended net across inductances that
            have a value lower than the threshold.
        capacitor_above : int, float, optional
            Threshold for the capacitor value. Search the extended net across capacitors that
            have a value higher than the threshold.
        exception_list : list, optional
            List of components to bypass when performing threshold checks. Components
            in the list are considered as serial components. The default is ``None``.
```

File: pyaedt/edb_core/net_class.py [hunk 4]
```diff
@@ -212,3 +282,44 @@ def create(self, name, net_p, net_n):
         diff_pair.api_create(name)._api_set_differential_pair(net_p, net_n)
 
         return self.items[name]
+
+    @pyaedt_function_handler
+    def auto_identify(self, positive_differentiator="_P", negative_differentiator="_N"):
+        """Generate diffperential pairs.
```
Comment (PipKat) [inline pyaedt/edb_core/net_class.py:288]: This needs to be rewritten in plain English. The Returns section also needs to have a description added.

File: pyaedt/edb_core/net_class.py [hunk 5]
```diff
@@ -212,3 +282,44 @@ def create(self, name, net_p, net_n):
...
+
+        Parameters
+        ----------
+        positive_differentiator: str, optional
+            Differentiator of the positive net.
+        negative_differentiator: str, optional
+            Differentiator of the negative net.
+        Returns
```
Comment (PipKat) [inline pyaedt/edb_core/net_class.py:296]: Add the Return info.

File: pyaedt/edb_core/nets.py [hunk 6]
```diff
@@ -246,7 +246,8 @@ def generate_extended_nets(
             Threshold of capacitor value. Search extended net across capacitors which has value higher than the
             threshold.
         exception_list : list, optional
-            List of components which bypass threshold check. The default is ``None``.
+            List of components which bypass threshold check. Components in the list are considerred as serial
+        components.The default is ``None``.
```
Comment (PipKat) [inline pyaedt/edb_core/nets.py:250]:
```suggestion
            List of components to bypass when performing threshold checks. Components
            in the list are considered as serial components. The default is ``None``.
```
The four preceding descriptions should be changed to match the suggestions made earlier for these same parameters.

File: pyaedt/edb_core/net_class.py [hunk 7]
```diff
@@ -212,3 +282,44 @@ def create(self, name, net_p, net_n):
...
+        -------
+
+        Examples
+        --------
+        >>> from pyaedt import Edb
+        >>> edbapp = Edb("myaedbfolder", edbversion="2021.2")
```
Comment (MaxJPRey) [inline pyaedt/edb_core/net_class.py:302]:
```suggestion
        >>> edbapp = Edb("myaedbfolder", edbversion="2023.1")
```

File: pyaedt/edb_core/net_class.py [hunk 8]
```diff
@@ -212,3 +284,45 @@ def create(self, name, net_p, net_n):
         diff_pair.api_create(name)._api_set_differential_pair(net_p, net_n)
 
         return self.items[name]
+
+    @pyaedt_function_handler
+    def auto_identify(self, positive_differentiator="_P", negative_differentiator="_N"):
+        """Generate diffperential pairs.
```
Comment (MaxJPRey) [inline pyaedt/edb_core/net_class.py:290]:
```suggestion
        """Generate differential pairs.
```

File: pyaedt/edb_core/net_class.py (hunk 4 above)
Comment (ring630) [reply pyaedt/edb_core/net_class.py]: @PipKat rewrote. Please help to check.

File: pyaedt/edb_core/net_class.py (hunk 5 above)
Comment (ring630) [reply pyaedt/edb_core/net_class.py:299]: @PipKat done.

**Review by svandenb-dev — APPROVED** [review]
LGTM
